common/inputs.js

```diff
@@ -172,5 +172,126 @@
                 }
             }
         }
+    }])
+
+    .directive('inputSwitch', ['dataFormats', 'integerLimits', 'maskOptions', 'modalBox', 'modalUtils', 'UriUtils', '$log', '$rootScope', function(dataFormats, integerLimits, maskOptions, modalBox, modalUtils, UriUtils, $log, $rootScope) {
+        return {
+            restrict: 'E',
+            templateUrl:  UriUtils.chaiseDeploymentPath() + 'common/templates/inputs/inputSwitch.html',
+            scope: {
+                column: '=',
+                displayType: '=',
+                model: '=?',
+                submissionRow: '=?',
+                foreignKeyData: '=?'
+            },
+            link: function(scope, elem, attr) {
+                scope.model = {};
+                scope.dataFormats = dataFormats;
+                scope.maskOptions = maskOptions;
+                scope.fileExtensionTypes = function () {
+                    return scope.column.filenameExtFilter.join(", ");
+                }
+
+                scope.int2min = integerLimits.INT_2_MIN;
+                scope.int2max = integerLimits.INT_2_MAX;
+                scope.int4min = integerLimits.INT_4_MIN;
+                scope.int4max = integerLimits.INT_4_MAX;
+                scope.int8min = integerLimits.INT_8_MIN;
+                scope.int8max = integerLimits.INT_8_MAX;
+
+                scope.booleanValues = ['', true, false];
+
+                if (scope.displayType === "timestamp") {
+                    scope.model.value = {
+                        date: null,
+                        time: null,
+                        meridiem: 'AM'
+                    }
+                } else if (scope.displayType === "file") {
+                    scope.model.value = {};
+                }
+
+                scope.getDisabledInputValue = function () {
+                    try {
+                        var disabled = scope.column.getInputDisabled("entry/create");
+                        if (disabled) {
+                            if (typeof disabled === 'object') return disabled.message;
+                            return '';
+                        } else if (scope.column.isForeignKey) {
+                            return 'Select a value';
+                        } else if (scope.column.isAsset) {
+                            return "No file Selected";
+                        }
+                    } catch (e) {
+                        $log.info(e);
+                    }
+                }
+
+                // Assigns the current date or timestamp to inputValue
+                scope.applyCurrentDatetime = function() {
```

Comment:
Are these functions that you have defined in this directive, the same functions that are currently in the recordedit page? Since at some point we might refactor recordedit to use this, would it help to add some TODOs to these functions in `form.controller.js`? 
If we want to change any of these functions, then we have to make sure that we're going to change the two identical places. So it might be easier to have a reminder that these functions are the same.